src/main/java/build/buildfarm/worker/ExecuteActionStage.java

```diff
@@ -137,8 +124,8 @@ protected void iterate() throws InterruptedException {
     Thread executorThread = new Thread(() -> executor.run(limits));
 
     synchronized (this) {
-      executors.add(executorThread);
-      size = executorClaims.addAndGet(limits.cpu.claimed);
+      slots.jobs.add(executorThread);
+      int size = getSlotUsage();
```

Comment:
Don't you need an executionSlotUsage.set here?

Reply:
We could.  Currently we only capture the metric on slot release.  This was probably done to avoid computing the metric twice as much.  InputFetchStage does the same thing.